modules/terminate-agent-hook/lambda/lambda_function.py

```diff
@@ -270,7 +434,7 @@ def handler(event, context):
             "Message": "No instances to terminate."
         }))
 
-    remove_unused_ssh_key_pairs(client=client, executor_name_part=os.environ['NAME_EXECUTOR_INSTANCE'])
+    remove_unused_ssh_key_pairs(client=ec2_client, executor_name_part=os.environ['NAME_EXECUTOR_INSTANCE'])
```

Comment:
note: as far as I believe, we can remove the SSH keys immediately. They are copied during startup into local files on the GitLab Runner instance and thus no longer needed.

Reply:
I have this line after the graceful terminate section to prevent it from running multiple time, since the graceful terminate section of the code is allowed to fail and retry.